data_analysis/static_gtfs_analysis.py

```diff
@@ -0,0 +1,402 @@
+# Goals:
+#
+# * Make a way to calculate the scheduled number of current active trips given
+#  a date, time, and route.
+#     - Take datetime and find what services are active on that date
+#     - Find what trips run on those services + route
+#     - Find which of those trips are "in progress" per stop_times
+# * ~Output most common shape by route~
+
+
+# imports
+from __future__ import annotations
+import os
+from pathlib import Path
+from dataclasses import dataclass
+
+import logging
+import calendar
+import pandas as pd
+import zipfile
+import requests
+import pendulum
+from io import BytesIO
+import shapely
+import geopandas
+
+from tqdm import tqdm
+
+VERSION_ID = "20220718"
+BUCKET = os.getenv('BUCKET_PUBLIC', 'chn-ghost-buses-public')
+
+logger = logging.getLogger()
+logging.basicConfig(level=logging.INFO)
+
+
+@dataclass
+class GTFSFeed:
+    """Class for storing GTFSFeed data.
+    """
+    stops: pd.DataFrame
+    stop_times: pd.DataFrame
+    routes: pd.DataFrame
+    trips: pd.DataFrame
+    calendar: pd.DataFrame
+    calendar_dates: pd.DataFrame
+    shapes: pd.DataFrame
+
+    @classmethod
+    def extract_data(cls, gtfs_zipfile: zipfile.ZipFile) -> GTFSFeed:
+        """Load each text file in zipfile into a DataFrame
+
+        Args:
+            gtfs_zipfile (zipfile.ZipFile): Zipfile downloaded from
+                CTA transit feeds e.g.
+                https://transitfeeds.com/p/chicago-transit-authority/
+                165/20220718/download"
+
+        Returns:
+            GTFSFeed: A GTFSFeed object containing multiple DataFrames
+                accessible by name.
+        """
+        logging.info(f"Extracting data from {gtfs_zipfile}")
+        data_dict = {}
+        pbar = tqdm(cls.__annotations__.keys())
+        for txt_file in pbar:
+            pbar.set_description(f'Loading {txt_file}.txt')
+            try:
+                with gtfs_zipfile.open(f'{txt_file}.txt') as file:
+                    df = pd.read_csv(file, dtype="object")
+                    logger.info(f'{txt_file}.txt loaded')
+
+            except KeyError as ke:
+                logger.info(f"{gtfs_zipfile} is missing required file")
```

Comment:
I like your refactor here! Just one very pedantic note, which doesn't really matter for our use case: part of the reason the original was set up how it was is that technically shapes, calendar, and calendar_dates are not actually fully required in the [GTFS specification](https://gtfs.org/schedule/reference/). (Because the rest of the script requires them I think it does make sense to call them required in our sense, but I just wanted to explain why some files were handled differently in the notebook... I was halfheartedly trying to make it GTFS-generic.)

Reply:
Thanks for the clarification and sharing the link to the GTFS specification. It's a useful guide for understanding the data in the zip file.